### PR TITLE
feature/CLS2-418-investment-project-get-add-endpoints

### PR DESCRIPTION
**Pull request by chopkinsmade**
### Description of change

- Add endpoints for POST, GET and LIST for an investment project task
- Moved the `task` and `id` properties from `InvestmentProjectTask` into a base model `BaseTaskType`. This is for the next tickets which will be adding a `CompanyTask` that need the same fields.

### Checklist

* [ ] Has this branch been rebased on top of the current `main` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [ ] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/main/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: datahub/task/models.py [hunk 1]
```diff
@@ -33,16 +33,23 @@ def __str__(self):
...
-    """Representation as an investment project task"""
-
+class BaseTaskType(BaseModel):
     id = models.UUIDField(primary_key=True, default=uuid.uuid4)
+
     task = models.ForeignKey(
         Task,
         on_delete=models.CASCADE,
-        related_name='investment_project',
+        related_name='%(app_label)s_%(class)s',
```
Comment (marijnkampf) [inline datahub/task/models.py:42]: How does this handle app labels and classes that are not pluralised by adding an 's'?

Comment (CaitBarnard) [reply datahub/task/models.py:42]: I don't think it does add an s, this confused me too and Chris showed me an example which didn't pluralise. I might have to retest this to understand it again

Comment (marijnkampf) [reply datahub/task/models.py:42]: I think this is some Django magic that is happening that I wasn't aware of.

**Review by marijnkampf — APPROVED** [review]

File: datahub/task/models.py (hunk 1 above)
Comment (CaitBarnard) [reply datahub/task/models.py:42]: We've had a look at it and can't work it out either, Django magic is often to blame

**Pull request merged**